### PR TITLE
integrations/operator: add tbot sidecar tests

### LINKED ISSUE #29006: Add operator sidecar test coverage
Operator sidecar logic is untested and this caused: https://github.com/gravitational/teleport/issues/28997

We should add coverage, for example:

- with sidecar integration tests (only spin up Teleport, no Kubernetes)
- with end to end tests (spin up Teleport, Kubernetes and apply manifests from yaml)

### PR DESCRIPTION
**Pull request by hugoShaka**
Fixes https://github.com/gravitational/teleport/issues/29006

Tests for https://github.com/gravitational/teleport/pull/30978

### CODE REVIEW

File: integrations/operator/sidecar/tbot_test.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+		Log:         logrus.StandardLogger(),
+	})
+
+	rcConf := servicecfg.MakeDefaultConfig()
+	rcConf.DataDir = t.TempDir()
+	rcConf.Auth.Enabled = true
+	rcConf.Proxy.Enabled = true
+	rcConf.Proxy.DisableWebInterface = true
+	rcConf.SSH.Enabled = true
+	rcConf.Version = "v2"
```
Comment (marcoandredinis) [inline integrations/operator/sidecar/tbot_test.go:144]: v3 is the latest

Comment (marcoandredinis) [inline integrations/operator/sidecar/tbot_test.go:144]: Is there a test helper function we can re-use here?

**Review by marcoandredinis — COMMENTED** [review]
Should the test fail on commits that precede this PR? https://github.com/gravitational/teleport/pull/30978

I did the following:
`git checkout d97200875509b729f13b342954c16e0d7a7dcb67` Commit just before the fix was merged
`git cherry-pick 200ec71c305a54749a1d39f934676f2bff21712d` cherry picked this PRs
And then ran the test
```
go test -timeout 30s -run ^TestTbotJoinTwice$ github.com/gravitational/teleport/integrations/operator/sidecar

ok  	github.com/gravitational/teleport/integrations/operator/sidecar	1.437s
```

I was expecting it to fail here

**Comment by hugoShaka** [comment]
> Should the test fail on commits that precede this PR? #30978

The test should not fail on `master` because @strideynet improved the tbot joining process which is more resilient. The test fails on `branch/v13` prior to the backport fix (you can try on `v13.3.5`, a few function signature changes must be adapted). The `v13.3.5` failure is not flaky (because the bug was a race), so you might want to run it multiple times. I've observed a 50% error rate on my M1 mac.

File: integrations/operator/sidecar/tbot_test.go (hunk 1 above)
Comment (hugoShaka) [reply integrations/operator/sidecar/tbot_test.go:144]: Yes, but it's in `testlib` which depends on `sidecar`, I'll break the circular dependency and use it.

**Comment by hugoShaka** [comment]
Superseded by https://github.com/gravitational/teleport/pull/34482#pullrequestreview-1744656194

**Pull request closed without merging**